Queryable/Queryable/View/SearchResultsView.swift

```diff
@@ -256,13 +257,23 @@ struct UpdateIndexView: View {
             .accessibilityAddTraits(.isLink)
             .accessibilityHint(Text("Click to build index for new photos to make them searchable"))
             .foregroundColor(Color.weakgreen)
+            .limitedLibraryPicker(isPresented: $showLimitedLibraryPicker)
             .onTapGesture {
-                goToIndexView = true
+                if PHPhotoLibrary.authorizationStatus(for: .readWrite) == .limited {
+                    showLimitedLibraryPicker = true
+                } else {
+                    goToIndexView = true
+                }
+            }
+            .onChange(of: showLimitedLibraryPicker) { showLimitedLibraryPicker in
```

Comment:
This is a bit of a confusing state machine, but I don't know of a better way to represent this here.

If the authorization is limited, instead of immediately going to index view, show the picker. Then, when the picker is dismissed, refetch the photos and go to the index view.